### PR TITLE
docs: describe rich parameters

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/6074

Preview: [link](https://github.com/coder/coder/blob/2a4f546155f6a0e7de29e00528778a72171902f4/docs/templates/parameters.md)

This PR adds a doc page describing rich parameters.

### CODE REVIEW

**Comment by mtojek** [comment]
Back to draft, I need to check one thing...

File: docs/templates/parameters.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+```hcl
+data "coder_parameter" "account_name" {
+  name        = "Account name"
+  description = "Cloud account name"
+  mutable     = true
+}
+```
+
+If a parameter contains the `default` property, coder will use it when the workspace user doesn't specify the custom value:
```
Comment (johnstcn) [inline docs/templates/parameters.md:93]:
```suggestion
If a parameter contains the `default` property, Coder will use this value 
 if the user does not specify any:
```

File: docs/templates/parameters.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+  legacy_variable = var.legacy_cpu
+}
+```
+
+Once users update their workspaces to the new template revision with rich parameters, the template admin can remove legacy variables, and strip `legacy_variable` properties.
+
+### Managed Terraform variables
+
+As parameters are intended to be used only for workspace customization purposes, Terraform variables can be freely managed by the admin to build templates. Workspace users are not able to modify
```
Comment (johnstcn) [inline docs/templates/parameters.md:223]:
```suggestion
As parameters are intended to be used only for workspace customization purposes, Terraform variables can be freely managed by the template author to build templates. Workspace users are not able to modify
```

File: docs/templates/parameters.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
+# Parameters (alpha)
+
+> Parameters are an [alpha feature](../contributing/feature-stages.md#alpha-features). See the [Rich Parameters Milestone](https://github.com/coder/coder/milestone/11) for more details.
+
+Templates can contain _parameters_, which prompt the user for additional information in the "create workspace" screen.
```
Comment (johnstcn) [inline docs/templates/parameters.md:5]:
```suggestion
Templates can contain _parameters_, which allow prompting the user for additional information when creating workspaces in both the UI and CLI.
```

File: docs/templates/parameters.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+}
+```
+
+## Legacy
+
+Prior to Coder v0.16.0 (Jan 2023), parameters were defined via Terraform `variable` blocks. These "legacy parameters" can still be used in templates, but will be removed in April 2023.
+
+```hcl
+variable "use_kubeconfig" {
+  sensitive   = true # Admin (template-level) parameter
```
Comment (johnstcn) [inline docs/templates/parameters.md:173]:
```suggestion
  sensitive   = true # Template-level parameter (not editable when creating a workspace)
```

File: docs/templates/parameters.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  type        = bool
+  description = <<-EOF
+  Use host kubeconfig? (true/false)
+  EOF
+}
+
+variable "cpu" {
+  sensitive   = false # User (workspace-level) parameter
```
Comment (johnstcn) [inline docs/templates/parameters.md:181]:
```suggestion
  sensitive   = false # User parameter (may be edited when creating a workspace)
```

File: docs/templates/parameters.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+```
+
+Once users update their workspaces to the new template revision with rich parameters, the template admin can remove legacy variables, and strip `legacy_variable` properties.
+
+### Managed Terraform variables
+
+As parameters are intended to be used only for workspace customization purposes, Terraform variables can be freely managed by the admin to build templates. Workspace users are not able to modify
+template variables.
+
+The admin user can enable managed Terraform variables mode by specifying the following flag:
```
Comment (johnstcn) [inline docs/templates/parameters.md:226]:
```suggestion
The template author can enable managed Terraform variables mode by specifying the following flag:
```

**Review by johnstcn — APPROVED** [review]
Some wording suggestions, in particular "admin" -> "template author".
Looks good otherwise!

**Pull request merged**